vectordb_orm/base.py

```diff
@@ -0,0 +1,178 @@
+from vectordb_orm.fields import EmbeddingField, VarCharField, BaseField, PrimaryKeyField
+from typing import Any
+import numpy as np
+from dataclasses import dataclass
+
+
+class MilvusBaseMeta(type):
+    def __getattr__(cls, name):
+        if name in set(cls.__annotations__.keys()):
+            return AttributeCompare(cls, name)
+        raise AttributeError(f"'{cls.__name__}' object has no attribute '{name}'")
+
+
+class MilvusBase(metaclass=MilvusBaseMeta):
+    _type_configuration: dict[str, BaseField]
+
+    def __init__(self, *values: Any, **data: Any) -> None:
+        if values:
+            raise ValueError("Use keyword arguments to initialize a Milvus object.")
+
+        allowed_keys = set(self.__class__.__annotations__.keys())
+
+        for key, value in data.items():
+            if key not in allowed_keys:
+                raise ValueError(f"Unexpected keyword argument '{key}'")
+
+        for key in allowed_keys:
+            if key in data:
+                setattr(self, key, data[key])
+            else:
+                field_configuration = self._type_configuration.get(key)
+                if field_configuration:
+                    setattr(self, key, field_configuration.default)
+                else:
+                    raise ValueError(f"Missing required argument '{key}'")
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__()
+
+        # Cache the type configurations and then remove the values. We want accessing the property of the
+        # class to return the value of the field (or default to AttributeCompare via __getattr__ in the case of
+        # a direct class attribute), not the field configuration itself.
+        cls._type_configuration = {}
+        for key in cls.__annotations__.keys():
+            attribute_value = getattr(cls, key)
+            cls._type_configuration[key] = attribute_value if isinstance(attribute_value, BaseField) else None
+            delattr(cls, key)
+
+    @classmethod
+    def collection_name(self) -> str:
+        if not hasattr(self, '__collection_name__'):
+            raise ValueError(f"Class {self.__name__} does not have a collection name, specify `__collection_name__` on the class definition.")
+        return self.__collection_name__
+
+    @classmethod
+    def embedding_dim(cls) -> int:
+        return cls.__embedding_dim__
+
+    @classmethod
+    def _create_collection(cls, milvus_client: Milvus):
+        cls._assert_has_primary()
+
+        fields: list[FieldSchema] = []
+
+        for attribute_name, type_hint in cls.__annotations__.items():
+            fields.append(
+                cls._field_schema_from_typehint(
+                    attribute_name,
+                    type_hint,
+                    cls._type_configuration.get(attribute_name)
+                )
+            )
+
+        schema = CollectionSchema(fields=fields, description=f"{cls.__name__} vectordb-generated collection")
```

Comment:
it could be good to expose the `__repr__` of the `Collection` or `CollectionSchema` somewhere- the user can also pull it by calling `str(collection)` or `str(collection._schema) `

[Here](https://github.com/milvus-io/pymilvus/blob/a1e3734dada1644e4ed0774e1408b8bf74836732/pymilvus/orm/collection.py#L129) and [here](https://github.com/milvus-io/pymilvus/blob/a1e3734dada1644e4ed0774e1408b8bf74836732/pymilvus/orm/schema.py#L76) are the definitions for them, they seems useful